### PR TITLE
CSP-105-Outer-ConfirmRegulatedStandard

### PR DESCRIPTION
**Pull request by raviudari01**

### CODE REVIEW

File: src/SFA.DAS.Roatp.CourseManagement.Api/Controllers/ProviderCourseEditController.cs [hunk 1]
```diff
@@ -34,5 +35,21 @@ public async Task<IActionResult> UpdateProviderCourseContactDetails(int ukprn, i
             }
             return NoContent();
         }
+
+        [HttpPost]
+        [Route("providers/{ukprn}/courses/{larsCode}/update-confirm-regulated-standard")]
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Api/Controllers/ProviderCourseEditController.cs:40]: The endpoint has two verbs in it, change to `update-approved-by-regulator`

**Review by DevenKShah — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Commands/UpdateApprovedByRegulator/UpdateApprovedByRegulatorCommand.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using MediatR;
+using System.Net;
+
+namespace SFA.DAS.Roatp.CourseManagement.Application.Standards.Commands.UpdateApprovedByRegulator
+{
+    public class UpdateApprovedByRegulatorCommand : IRequest<HttpStatusCode>
+    {
+        public int Ukprn { get; set; }
+        public int LarsCode { get; set; }
+        public bool? IsApprovedByRegulator { get; set; }
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Commands/UpdateApprovedByRegulator/UpdateApprovedByRegulatorCommand.cs:10]: This cannot be null. As the user is trying to update the value, we should never allow null here.

**Review by DevenKShah — CHANGES_REQUESTED** [review]
One small change required.

**Review by DevenKShah — APPROVED** [review]

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.Roatp.CourseManagement/InnerApi/Requests/UpdateProviderCourse.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿namespace SFA.DAS.Roatp.CourseManagement.InnerApi.Requests
+{
+    public class UpdateProviderCourse
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement/InnerApi/Requests/UpdateProviderCourse.cs:3]: ProviderCourseUpdateModel

File: src/SFA.DAS.Roatp.CourseManagement.UnitTests/Application/Standards/Commands/UpdateContactDetailsCommandHandlerTests.cs [hunk 4]
```diff
@@ -22,13 +24,28 @@ public class UpdateContactDetailsCommandHandlerTests
...
-            apiClientMock.Setup(a => a.PostWithResponseCode<UpdateContactDetailsRequest>(It.IsAny<UpdateContactDetailsRequest>())).ReturnsAsync(new ApiResponse<UpdateContactDetailsRequest>(request, HttpStatusCode.NoContent, string.Empty));
+            apiClientMock.Setup(a => a.GetWithResponseCode<GetProviderCourseResponse>(It.IsAny<GetProviderCourseRequest>())).ReturnsAsync(new ApiResponse<GetProviderCourseResponse>(apiResponse, HttpStatusCode.OK, string.Empty));
+
+            apiClientMock.Setup(a => a.PostWithResponseCode<UpdateProviderCourseRequest>(It.IsAny<UpdateProviderCourseRequest>())).ReturnsAsync(new ApiResponse<UpdateProviderCourseRequest>(request, HttpStatusCode.NoContent, string.Empty));
 
             await sut.Handle(command, cancellationToken);
 
-            apiClientMock.Verify(a => a.PostWithResponseCode<UpdateContactDetailsRequest>(It.IsAny<UpdateContactDetailsRequest>()), Times.Once);
+            apiClientMock.Verify(a => a.PostWithResponseCode<UpdateProviderCourseRequest>(It.IsAny<UpdateProviderCourseRequest>()), Times.Once);
+        }
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.UnitTests/Application/Standards/Commands/UpdateContactDetailsCommandHandlerTests.cs:37]: Tests missing for conversion from Command to Request model. Same set of tests should be added to other handler as well.

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**